### PR TITLE
Remove goconvey and godep

### PR DESCRIPTION
**Pull request by jharlap**
The community guideline is gravitating towards: vendor for applications, not for libraries. This PR changes the remaining goconvey tests into stdlib and testify/assert tests, and removes godeps in the process.

### CODE REVIEW

**Comment by jharlap** [comment]
Note that jenkins fails while travis passes. Have to get Jenkins disabled for this repo, because it's so much easier to use travis for open source Go packages ;)

File: sqs_interface_test.go [hunk 1]
```diff
@@ -5,11 +5,9 @@ import (
...
+	"github.com/stretchr/testify/assert"
 )
 
-func TestSQSInterfaceMatches(t *testing.T) {
-	Convey("sqs.New() should implement SQSAPI", t, func() {
-		So(sqs.New(nil), ShouldImplement, (*sqsconsumer.SQSAPI)(nil))
-	})
+func TestSQSInterfaceImplementsSQSAPI(t *testing.T) {
+	assert.Implements(t, (*sqsconsumer.SQSAPI)(nil), sqs.New(nil))
 }
```
Comment (minosabaki) [inline sqs_interface_test.go]: While I know you are just doing a conversion....this test is a little weird.

Comment (jharlap) [inline sqs_interface_test.go]: How so? It exists simply to ensure that if the AWS library changes the SQS API then the local interface will not match and the test will start failing.

**Comment by minosabaki** [comment]
R+ :shipit:

**Comment by jharlap** [comment]
Disabled travis, fixed race conditions in tests which were causing circle to fail (because it runs with the race detector enabled, which is good but...).

Also exposed WaitSeconds so that test suites using sqsconsumer can reduce the wait time from the default of 20s to 1s

**Pull request merged**